test/test_example.py

```diff
@@ -55,6 +55,23 @@ def test_list_extracted_into_arguments(self, first_value, second_value):
     def test_dicts_extracted_into_kwargs(self, first, second, third):
         self.assertTrue(first < third < second)
 
+    @data(1, 2, 3)
+    @data(4, 5, 6)
+    def test_products(self, first_value, second_value):
+        self.assertTrue(first_value < second_value)
```

Comment:
How about testing for the number of products expected? It'd be nice if these three tests also told someone perusing the code what to expect as far as what products result from the nesting.